modules/stochastic_tools/src/samplers/ParallelSubsetSimulation.C

```diff
@@ -178,13 +182,26 @@ ParallelSubsetSimulation::computeSample(dof_id_type row_index, dof_id_type col_i
   else
   {
     const dof_id_type loc_ind = row_index - getLocalRowBegin();
-    const Real rv = Normal::quantile(getRand(seed_value), _markov_seed[col_index][loc_ind], 1.0);
-    const Real acceptance_ratio = std::log(Normal::pdf(rv, 0, 1)) -
-                                  std::log(Normal::pdf(_markov_seed[col_index][loc_ind], 0, 1));
-    const Real new_sample = acceptance_ratio > std::log(getRand(seed_value + 1))
-                                ? rv
-                                : _markov_seed[col_index][loc_ind];
-    val = Normal::cdf(new_sample, 0, 1);
+
+    if (_sampling_method == "ComponentWiseMH")
+    {
+      const Real rnd1 = getRand(seed_value);
+      const Real rnd2 = getRand(seed_value + 1);
+      val = AdaptiveMonteCarloUtils::proposeNewSampleComponentWiseMH(
+          _markov_seed[col_index][loc_ind], rnd1, rnd2);
+    }
+
+    else if (_sampling_method == "MH")
+    {
+      const Real rnd1 = getRand(seed_value);
+      const Real rnd2 = getRand(seed_value + 1);
+      std::vector<Real> new_sample =
+          AdaptiveMonteCarloUtils::proposeNewSampleMH(_distributions, rnd1, _inputs, _inputs_sto);
```

Comment:
I don't see this quantity used anywhere??

Reply:
Thanks for the great catch!  In fact, `new_sample` is stored in _inputs_sto and that's why I called `_inputs_sto[col_index].back()` in the next line. I know this can be confusing, thanks for pointing it out. Changes are made in the newly pushed code.